tests/kani/MMIO/semantics.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+const BUFFER: *mut u32 = 0x8000 as *mut u32;
+
+#[cfg(kani)]
+#[kani::proof]
+#[kani::mmio_region(0x8000, 4)]
+fn test_read_after_write_doesnt_havoc() {
+    let val = 12;
+    unsafe {
+        core::ptr::write_volatile(BUFFER, val);
+        let new_val = core::ptr::read_volatile(BUFFER);
+        assert_eq!(new_val, val);
+    }
+}
+
+#[cfg(kani)]
+#[kani::proof]
+#[kani::mmio_region(0x8000, 4)]
+fn test_read_is_stable() {
+    unsafe {
+        let val1 = core::ptr::read_volatile(BUFFER);
+        let val2 = core::ptr::read_volatile(BUFFER);
+        assert_eq!(val1, val2);
+    }
+}
+
+//TODO Weirdly, these fail even tho the other tests pass???
```

Comment:
This looks like a blocker to me. Maybe we should investigate this behavior a bit better.